RabbitMQ.Stream.Client/IProducer.cs

```diff
@@ -16,13 +16,42 @@ namespace RabbitMQ.Stream.Client;
 
 public interface IProducer
 {
+    /// <summary>
+    /// Send the message to the stream in asynchronous mode.
+    /// The client will aggregate messages and send them in batches.
+    /// It works in most cases.
```

Comment:
What do we mean here by "it works in most cases" ?
Does it not work sometimes and I just lose messages? 😜 

I think it will be helpful to document how often does it send messages that it has accumulated, or how to configure how often it sends messages.